dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,334 @@
+
+        Returns:
+            np.ndarray:  a correlation matrix of shape (n_features, n_features). The diagonal is 1 if `y` is not passed, otherwise it is `corr(x_i, y)`.
+        """
+        # generate correlation matrix, use tempfile and chunked process
+
+        f_x = tempfile.NamedTemporaryFile()
+        f_corr = tempfile.NamedTemporaryFile()
+
+        X_memmapped = np.memmap(f_x, "float64", mode="w+", shape=X.shape)
+        X_memmapped[:] = X[:]
+        X_memmapped.flush()
+
+        correlation_matrix = np.memmap(
+            f_corr, "float64", mode="w+", shape=(X.shape[1], X.shape[1])
+        )
+        corrcoef(X_memmapped, out=correlation_matrix, copy=False, rowvar=False)
+
+        if y is None:
+            np.fill_diagonal(
+                correlation_matrix, [0 for _ in range(correlation_matrix.shape[1])]
+            )
+        else:
+            X_memmapped = (X_memmapped - X_memmapped.mean(axis=0)) / (
+                X_memmapped.std(axis=0)
+            )
+            y_standardized = (y - y.mean()) / y.std()
+            np.fill_diagonal(
+                correlation_matrix,
+                self.alpha * (1 / y.shape[0]) * (X_memmapped.T @ y_standardized),
+            )
+
+        f_x.close()
+
+        return correlation_matrix, f_corr
+
+    def calculate_mutual_information(self, X: np.ndarray, y: np.ndarray):
+        """Mutual information matrix with the outcome
+
+        Args:
+            X (np.ndarray): An array of shape (n_obserations, n_features)
+            y (np.ndarray): An array of shape (n_observations, 1) with the outcome variable
+
+        Raises:
+            NotImplementedError: This function is not yet implemented, so it will always raise an error
+        """
+        raise NotImplementedError("Mutual information is not yet implemented")
+
+    def fit(
+        self,
+        X: Union[np.ndarray, pd.DataFrame],
+        y: Union[np.ndarray, pd.DataFrame, pd.Series, None] = None,
+        number_of_features: int = None,
+        strict: bool = True,
+    ):
+        """Conducts the feature selection proceedure and finds the features to keep.
+
+        Args:
+            X (Union[np.ndarray, pd.DataFrame]): a matrix-like object where each row is an observation, and each column is a feature to be selected. If mutual information is selected
+            every non-binary row will be assumed to be continuous.
+            y (Union[np.ndarray, pd.DataFrame, pd.Series, None], optional): outcome to be considered in feature selection. If provided, the correlation with the outcome will be considered,
+            if not then only the covariances among the features will be considered. Required for mutual information. Defaults to None.
+            number_of_features (int, optional): The number of features to be selected. If `strict` is `True` exactly this number of features,
+            otherwise at most this number of features is selected. If `None` then `n_default_feature` is used. Defaults to None.
```

Comment:
```suggestion
            number_of_features: Number of features to be selected. If ``strict`` is ``True``, 
                exactly this number of features is selected; otherwise this is an upper bound. 
                If set to ``None``, ``n_default_feature`` features are selected.
```